src/software/ai/hl/stp/tactic/crease_defender/crease_defender_fsm.cpp

```diff
@@ -91,18 +106,39 @@ void CreaseDefenderFSM::blockThreat(
     TbotsProto::BallCollisionType ball_collision_type =
         TbotsProto::BallCollisionType::ALLOW;
     if ((event.common.world_ptr->ball().position() - destination).length() <
-        (event.common.robot.position() - destination).length())
+        (robot_position - destination).length())
     {
         ball_collision_type = TbotsProto::BallCollisionType::AVOID;
     }
 
+    AutoChipOrKick auto_chip_or_kick{AutoChipOrKickMode::OFF, 0};
+    auto goal_line_segment =
+        Segment(event.common.world_ptr->field().friendlyGoal().posXPosYCorner(),
+                event.common.world_ptr->field().friendlyGoal().posXNegYCorner());
+    Ray robot_shoot_ray = Ray(robot_position, robot_orientation);
+    std::vector<Point> goal_intersections =
+        intersection(robot_shoot_ray, goal_line_segment);
+    Stadium threat_zone = Stadium(
+        robot_position, Vector::createFromAngle(robot_orientation).normalize(1), 0.1);
```

Comment:
the 1 in `normalize(1)` and the `0.1` radius seems like it should be a constant somewhere. Also, 0.1 seems like a pretty small radius. I'm not sure about this, but I feel like we should atleast be using 0.25 or something. I suggest using the debug visualize layer in Tscope to draw the Stadium you're creating, and atleast visually verifying that its size and position makes sense.

https://github.com/UBC-Thunderbots/Software/blob/99e9e9e2a883d01122a377dbee8a12b7d50d659f/src/proto/message_translation/tbots_protobuf.h#L154-L170

Reply:
Ah good catch, I forgot to set those as constants after I tuned them the Visualization logger

Reply:
I still feel like 0.1 is a small radius. That's basically the radius of a robot. I think to be safe we should increase that to a larger value so we're more conservative on chipping the ball away in risky situations.

Reply:
It might also make more sense to make this a Polygon (using `Polygon::fromSegment`) instead of a Stadium. Since if we make the stadium radius bigger, it would also cover a larger area behind the crease defender as well, where as our goal is to cover only the front. Here's an example of how I achieved something similar, but with a Polygon: https://github.com/UBC-Thunderbots/Software/blob/668b9b77232426078aa418239293e4145264a79e/src/software/ai/hl/stp/tactic/dribble/dribble_fsm.cpp#L28-L31